App documentation from developers website (#316)
Esri/dynamic-situational-awareness-qt: doc/index.md

# Overview

## Tools

![](./images/dsa-ss-vehicle-start-anno.png "Diagram of DSA Vehicle UI at startup")<br>*Startup display (DSA Vehicle)*

![](./images/dsa-ss-handheld-start-anno.png "Diagram of DSA Handheld UI at startup")<br>*Startup display (DSA Handheld)*

### Navigation toolbar

The navigation toolbar orients your view of the map. The tool appears vertically on the display's right side. The navigation toolbar allows you to:

|  |  |
| :---: | --- |
| ![](./images/dsa-icon-menu-home-32.png "Return to startup") | Return viewpoint to startup location |
| ![](./images/dsa-icon-plus-minus-32.png "Zoom in/out") | Zoom in and out |
| ![](./images/dsa-icon-2d-32.png "Set 2D perspective") | Set viewpoint to 2D perspective (looking straight down) |
| ![](./images/dsa-icon-gps-off-32.png "GPS toggle") | Show own location as a moving symbol on the map |
| ![](./images/dsa-icon-rotate-mode-32.png "Rotate mode toggle") | Toggle map navigation control between panning and map rotation |

![](./images/dsa-v-nav-demo.gif "Navigation tools reorienting the map")<br>*Navigation tools orienting the map*

The navigation toolbar can be turned off in the [Settings panel](#settings-panel).

> <!-- ## ![](./images/dsa-icon-menu-home-32.png) Compass -->

Reviewer: Remove?

```suggestion
```

- Handheld app: Toolbar horizontally across the bottom of the display.

The tool categories include:

|  |  |
| :---: | --- |
| ![](./images/dsa-icon-dsa-32.png) | Return command bar to start |
| ![](./images/dsa-icon-map-tools-32.png) | [Map tools](#map-tools) (manage map elements) |
| ![](./images/dsa-icon-create-reports-32.png) | [Create report tools](#create-report-tools) (such as observation reports) |
| ![](./images/dsa-icon-analysis-tools-32.png) | [Analysis tools](#exploratory-visual-analysis) (viewshed, line of sight) |
|  ![](./images/dsa-icon-alert-tools-32.png)  | [Alert tools](#alerts-and-conditions) (proximity, geofence, attribute-driven) |
| ![](./images/dsa-icon-markup-tools-32.png)  | [Markup tools](#markup-tools) (sketching) |

## Settings panel

![](./images/dsa-tool-settings.png "Setting panel")<br>*Settings panel (DSA Vehicle)*

The settings panel accesses the app's About information, app settings, and app close (for desktop platforms).

- Vehicle app: Control is at the bottom of the tools category panel on the left of the display.
- Handheld app: Control is in the toolbar when the DSA category is selected.

> <!-- TODO - Verify the following paragraph: Some settings are persisted, but ALL of them? -->

Reviewer: Hopefully this already got verified. I think we can remove

```suggestion
```

When you change settings, they are persisted in the configuration file.

## Map tools

### ![](./images/dsa-icon-convert-xy-32.png) Convert X/Y

It is common for analysts to work across various systems that use different coordinate notation formats to pinpoint places on the Earth. Working with multiple organizations simultaneously, each may require or provide coordinate information in different notation formats. You may need to consume, translate, relay, and/or display those notations. For example, you may have a coordinate expressed as Decimal Degrees (DD) notation with five decimal places, and you want to use that location in another system that expects Military Grid Reference System (MGRS) notation with whitespace between the grid zone designator, the 100km square identifier, and the numerical eastings and northings.

The Convert X/Y tool accepts a location in any supported coordinate notation or directly from the map and provides that location in different notation. The tool lets you copy converted notations for use elsewhere in the app, such as in an Observation report.

![](./images/dsa-callout-coordconv.png "Convert XY")<br>*Convert X/Y Tool (DSA Handheld)*

> <!-- <img style="float: right;" src="./images/dsa-callout-coordconv.png" alt="Convert XY"> -->

Reviewer: Appears to be a hidden alternative to L137, so we can remove

```suggestion
```

Specify the folders the app searches for local data using your [App configuration settings](#app-configuration-settings). Local data that has been found on your device is listed by this tool. Any data supported by ArcGIS Runtime SDKs can be used, including:

- [GeoPackage]
- [Tile package] (TPK)
- [Mobile geodatabase]
- KML
- [Raster formats], including:
  - JPEG / JPEG 2000
  - PNG
  - IMG
  - Raster Product Format (RPF) - CADRG and CIB
  - National Imagery Transmission Format (NITF)
  - Digital Terrain Elevation Data (DTED)
  - [Mobile Mosaic Dataset]
  - MrSID

- [Shapefile] (SHP)
- [Scene Layer Package] (SLPK)

### ![](./images/dsa-icon-overlays-32.png) Overlays

![](./images/dsa-tool-overlays.png "Overlays")<br>*Overlays tool (DSA Vehicle)*

A set of overlays manages what is drawn on the map. Overlays may be reordered, zoomed to, or deleted. Manage overlays to present information on the map that is most important to you. Add new overlays with the [Add data](#add-data) tool.

![](./images/dsa-tool-overlays-menu.png "Overlays (context menu)")<br>*Overlays tool context menu (DSA Vehicle)*

### ![](./images/dsa-icon-basemaps-32.png) Basemaps

> <!-- <img style="float: left;" src="./images/dsa-tool-basemap.png" alt="Basemaps"> -->

Reviewer: Appears to be a hidden alternative to L198 so we can remove

```suggestion
```

Tools are categorized by workflows, shown on the toolbar. Select one of the workflows on the toolbar to select tools to be displayed on the command bar.

- Vehicle app: Toolbar appears vertically down the display's left side.
- Handheld app: Toolbar horizontally across the bottom of the display.

The tool categories include:

|  |  |
| :---: | --- |
| ![](./images/dsa-icon-dsa-32.png) | Return command bar to start |
| ![](./images/dsa-icon-map-tools-32.png) | [Map tools](#map-tools) (manage map elements) |
| ![](./images/dsa-icon-create-reports-32.png) | [Create report tools](#create-report-tools) (such as observation reports) |
| ![](./images/dsa-icon-analysis-tools-32.png) | [Analysis tools](#exploratory-visual-analysis) (viewshed, line of sight) |
|  ![](./images/dsa-icon-alert-tools-32.png)  | [Alert tools](#alerts-and-conditions) (proximity, geofence, attribute-driven) |
| ![](./images/dsa-icon-markup-tools-32.png)  | [Markup tools](#markup-tools) (sketching) |

## Settings panel

![](./images/dsa-tool-settings.png "Setting panel")<br>*Settings panel (DSA Vehicle)*

The settings panel accesses the app's About information, app settings, and app close (for desktop platforms).

> - Vehicle app: Control is at the bottom of the tools category panel on the left of the display.

Reviewer: Line 102 refers to this as a toolbar rather than a 'tools category panel' so maybe that's more appropriate here? (You can see this "Settings" gear icon in the line 57 screenshot.) Otherwise, it looks like the term "tools" is lacking a possessive apostrophe.

```suggestion
- Vehicle app: Control is at the bottom of the toolbar on the left of the display.
```

Author: To me it appears that "tools" is plural, not possessive. And with that, this name feels intentional and I would rather keep as-is.

### ![](./images/dsa-icon-viewshed-32.png) Viewshed

![](./images/dsa-tool-viewshed.png "Viewshed")<br>*Viewshed analysis (DSA Vehicle)*

[Viewshed analysis] is a type of visibility analysis that shows the visible and obstructed areas within a directed field of view. Viewshed analysis shows you the areas of the scene that are visible from a given observer, determined by the terrain (represented by an elevation surface), buildings and other 3D features (represented by scene layers, graphics, and so on), and the properties of the observer. A viewshed is adjustable to show what is visible within a specified distance in a certain direction or all directions. Areas that are visible from the location are highlighted in green, and not-visible areas are highlighted in red. As the location moves, the viewshed analysis is recalculated.

> <!-- TODO - Add details -->

Reviewer: This line can probably be removed. Likely accidentally left in during the original authoring.

## Alerts and conditions

Alerts are events that are triggered when certain conditions, or rules, are met against the real-time feeds and messages displayed in the app.  These conditions are defined by the user and constantly evaluated as new messages are received and updated.  Spatial conditions can be created, such as geofences ("within area of") or distance ("within distance of"), or attribute conditions, such as "status911 = true" or "speed < 50".

### New alert notification

The user is notified of new alerts on the Tool Categories bar. The number in the red circle indicates any new alerts that have been added to the Alerts View since it was last opened.

![](./images/dsa-tool-alert-triggered.png "New alerts indicator")<br>*New alerts indicator (DSA Handheld)*

### ![](./images/dsa-icon-view-alerts-32.png) Alerts view

![](./images/dsa-tool-alert-view-active.png "View active alerts")<br>*Alerts view tool (DSA Handheld)*

> Use the View tool to view a list of active alerts. As alerts become active or inactive the app updates the list. You can filter the list by priority, and zoom to, highlight or dismiss an individual view on the list.

Reviewer: Add comma for pause and Oxford comma:

```suggestion
Use the View tool to view a list of active alerts. As alerts become active or inactive, the app updates the list. You can filter the list by priority, and zoom to, highlight, or dismiss an individual view on the list.
```

### ![](./images/dsa-icon-alert-conditions-32.png) Conditions

![](./images/dsa-tool-alert-conditions.png "Conditions wizard")<br>*Alert conditions tool (DSA Handheld)*

> When you select the Conditions tool the app presents a list of the defined alert conditions. When you press CREATE NEW, a wizard asks for details about the conditions that will trigger the new alert.

Reviewer: ```suggestion
When you select the Conditions tool, the app presents a list of the defined alert conditions. When you press CREATE NEW, a wizard asks for details about the conditions that will trigger the new alert.
```

In particular, performing spatial analysis (for example, a geofence) against many moving entities can be computationally expensive. To help alleviate this cost, the `GeometryQuadtree` can be used to create a spatial look-up structure for working with multiple [Geometry] objects. The quadtree is built to cover the full extent (an [Envelope] object) of the geometry and each object is recursively assigned to a leaf or node of the tree up to a maximum depth. The maximum depth of the tree can be assigned at creation time - generally 8 offers a good trade-off between granularity and the time taken to build the tree. The tree is a sparse structure, that is, any nodes which contain no geometry are removed. Once built, this structure offers very fast lookup of the candidate geometries which may intersect with a given query geometry. For performance reasons, the tree uses bounding box intersection tests only. The results are returned as a list of geometry objects which can be used for exact intersection tests using the [GeometryEngine]. The quadtree will connect to changes to the underlying geometry objects and can also be updated to include new features.

> ***Developer tip*** Building the quadtree is the most expensive part of the operation so care should be taken to do this only when required. For example, the quadtree is a useful tool where there are many features which change infrequently (for example, a static feature layer) but would be less appropriate for a small number of constantly changing features (for example, your current location). For very large datasets, the cost to build the tree may be very high, so it may be worth moving it's construction to a background thread to avoid blocking the GUI thread.

Reviewer: it's --> its

```suggestion
***Developer tip*** Building the quadtree is the most expensive part of the operation so care should be taken to do this only when required. For example, the quadtree is a useful tool where there are many features which change infrequently (for example, a static feature layer) but would be less appropriate for a small number of constantly changing features (for example, your current location). For very large datasets, the cost to build the tree may be very high, so it may be worth moving its construction to a background thread to avoid blocking the GUI thread.
```

## Collaboration

You can use DSA to collaborate with teammates by sharing [observation reports](#observation-report) or [markup overlays](#markup-tools) over the peer-to-peer network.

### ![](./images/dsa-icon-create-reports-32.png) Create report tools

Create reports to inform teammates of activities or entities in the area. Currently, DSA supports creating observation reports.

### ![](./images/dsa-icon-observation-32.png) Observation report

When you select the observation report tool, a wizard asks for details to include:

- The reporting unit's name (default is your unit)
- The size of the observation
- The observed activity
- The observation location (default is your location)
- Description of the observation
- The date and time of the observation (default is current time)

***Developer tip:*** You can follow the create report patterns used by this app to create your own reports with their own purpose and attributes and share those reports with your teammates.

### ![](./images/dsa-icon-markup-tools-32.png) Markup tools

<!-- <img style="float: left;" src="./images/dsa-tool-markup.png" alt="Draw on screen"> -->

> <!-- ![](./images/dsa-tool-markup.png "Draw on screen") -->

Reviewer: I think these lines can be removed since they're hidden
